buildpacks/sbt/src/layers/sbt_boot.rs

```diff
@@ -79,16 +77,14 @@ fn get_layer_env_scope(available_at_launch: bool) -> Scope {
 pub(crate) struct SbtLayerMetadata {
     sbt_version: Version,
     layer_version: String,
-    stack_id: StackId,
 }
 
 const LAYER_VERSION: &str = "v1";
 
 impl SbtLayerMetadata {
-    fn current(layer: &SbtBootLayer, context: &BuildContext<SbtBuildpack>) -> Self {
+    fn current(layer: &SbtBootLayer) -> Self {
         SbtLayerMetadata {
             sbt_version: layer.for_sbt_version.clone(),
-            stack_id: context.stack_id.clone(),
```

Comment:
Now that the stack ID isn't part of the layer metadata, the cache won't be invalidated when the stack changes.

Are any parts of the SBT cache stack specific? eg what if dependencies are bindings against C libraries in the base images, will any of the compiled files be stack specific? Even if SBT is smart enough to rebuild those assets on the next run after the base image change, presumably the cache will be bloated by 2x the compiled assets now?

Also, does any of the above also apply to the maven/gradle component CNBs (which weren't invalidating on stack change even before this PR)? Should they be invalidating caches upon distro name/version changes?

Reply:
Yeah, this is currently inconsistent across all JVM CNBs. I filed https://github.com/heroku/buildpacks-jvm/issues/663 to track this.